inc/routes.php

```diff
@@ -13,58 +13,75 @@
  * @since 0.1.0
  */
 function theme_sniffer_endpoint_init() {
-	register_rest_route( 'wp/v2/theme-sniffer/', '/sniff-run', array(
-		'methods'  => 'POST',
+	register_rest_route( 'theme-sniffer/v1', '/sniff-run', array(
+		'methods'  => 'GET',
 		'callback' => 'theme_sniffer_run_sniffer',
 	) );
 
-	register_rest_route( 'wp/v2/theme-sniffer/', '/individual-sniff', array(
-		'methods'  => 'POST',
+	register_rest_route( 'theme-sniffer/v1', '/individual-sniff', array(
+		'methods'  => 'GET',
 		'callback' => 'theme_sniffer_individual_sniff',
 	) );
 }
 
 /**
  * Callback function for the run sniffer endpoint
  *
+ * @param \WP_REST_Request $request Full data about the request.
  * @since 0.1.0
  */
-function theme_sniffer_run_sniffer() {
+function theme_sniffer_run_sniffer( \WP_REST_Request $request ) {
+	$headers = $request->get_headers();
+
 	// Bail if empty.
-	if ( empty( $_POST['themename'] ) ) {
-		return;
+	if ( empty( $_GET['themeName'] ) ) {
+		$message = esc_html__( 'Theme name not selected.', 'theme-sniffer' );
+		$error = new WP_Error( '-1', $message );
+		wp_send_json_error( $error );
 	}
 
 	// Verify nonce.
-	if ( ! isset( $_POST['theme_sniffer_nonce'] ) || ! wp_verify_nonce( $_POST['theme_sniffer_nonce'], 'wp_rest' ) ) {
-		esc_html_e( 'Nonce error', 'theme-sniffer' );
-		return;
+	if ( ! wp_verify_nonce( $headers['x_wp_nonce'][0], 'wp_rest' ) ) {
+		$message = esc_html__( 'Nonce error.', 'theme-sniffer' );
+		$error = new WP_Error( '-1', $message );
+		wp_send_json_error( $error );
 	}
 
+	// Exit if plugin not bundled properly.
 	if ( ! file_exists( THEME_SNIFFER_DIR . '/vendor/autoload.php' ) ) {
-		$message = sprintf( esc_html__( 'It seems you are using GitHub provided zip for the plugin. Visit %1$sInstalling%2$s to find the correct bundled plugin zip.', 'theme-sniffer' ), '<a href="https://github.com/ernilambar/theme-sniffer#installing" target="_blank">', '</a>' );
+		// translators: Placeholders are used for inserting hardcoded links to the repository.
+		$message = sprintf( esc_html__( 'It seems you are using GitHub provided zip for the plugin. Visit %1$sInstalling%2$s to find the correct bundled plugin zip.', 'theme-sniffer' ), '<a href="https://github.com/WPTRT/theme-sniffer#installing" target="_blank">', '</a>' );
 		$error = new WP_Error( '-1', $message );
 		wp_send_json_error( $error );
 	}
 
-	$theme_slug = esc_html( $_POST['themename'] );
+	if ( empty( $_GET['wpRulesets'] ) ) {
+		$message = esc_html__( 'Please select at least one standard.', 'theme-sniffer' );
+		$error = new WP_Error( '-1', $message );
+		wp_send_json_error( $error );
+	}
 
-	if ( isset( $_POST['hide_warning'] ) && 'true' === $_POST['hide_warning'] ) {
-		$args['show_warnings'] = true;
+	$theme_slug = sanitize_text_field( wp_unslash( $_GET['themeName'] ) );
+
+	if ( isset( $_GET['hideWarning'] ) && 'true' === $_GET['hideWarning'] ) {
+		$args['show_warnings'] = 'false'; // MANUAL INSPECTION!!!!!!!
```

Comment:
What is the reason for the code comment?

Reply:
Oh, my bad, forgot to remove this comment. I needed to check these `$_GET` variables. Will remove this.